tests/test_core.py

```diff
@@ -425,6 +425,44 @@ def __test_bad_tuple_list(tuple_list):
     __test_bad_tuple_list(badargs)
 
 
+def test_ensure_valid_source_time_tuple():
```

Comment:
Don't forget to update the name of this test function once we settle on a name for the func it's testing :)

Reply:
Will do!